src/components/shared/Button/index.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable react/button-has-type */
+import { Component } from 'react';
+import classNames from 'classnames/bind';
+import styles from './style.css';
+
+const cx = classNames.bind(styles);
+
+export default class Button extends Component {
+  render() {
+    const { className, children, ...props } = this.props;
+    const buttonClass = cx({
+      Button: true,
+      [`${className}`]: true,
+    });
+
+    return (
+      <button className={buttonClass} {...props}>
+        {children}
+      </button>
```

Comment:
1. children 그대로 넘겨주면 되겠는데요?
2. className에 아무 것도 안 넘긴다면?

```suggestion
    const { className, ...props } = this.props;
    const buttonClass = cx(className, Button);

    return (
      <button className={buttonClass} {...props} />
```

Reply:
앗 조건부 아니고 항상 넣어줄거면 심플하게 인자로 넣어주면 되는거였군요! [리드미](https://github.com/JedWatson/classnames#classnames)에 Syntax 다시 읽고왔습니당ㅎㅎ 감사합니다!
```js
classNames('foo', 'foo', 'bar'); // => 'foo bar'
classNames('foo', { foo: false, bar: true }); // => 'bar'
```

Reply:
말씀해주신 것처럼 `children`을 `...props`에 한꺼번에 넣을 경우 JSX가 어떻게 처리되는지 아직 머릿 속에 잘 안그려져서 Babel에 적어봤습니다..!

```js
// children을 따로 빼서 넘겨주는 경우
<button className={buttonClass} {...props}>
  {children}
</button>
```
```js
// 트랜스파일 후
export default class Button extends Component {
  render() {
    const { className, children, ...props } = this.props;
    const buttonClass = cx(className, "Button");
    return /*#__PURE__*/ React.createElement(
      "button",
      _extends(
        {
          className: buttonClass
        },
        props
      ),
      children
    );
  }
}
```


```js
// children을 ...props에 포함시켜넣거나 props와 같은 위계(?)에서 넘겨주는 경우
<button className={buttonClass} children={children} {...props} />
```
```js
// 트랜스파일 후
export default class Button extends Component {
  render() {
    const { className, children, ...props } = this.props;
    const buttonClass = cx(className, "Button");
    return /*#__PURE__*/ React.createElement(
      "button",
      _extends(
        {
          className: buttonClass,
          children: children
        },
        props
      )
    );
  }
}
```
두 경우에 트랜스파일링 후 children이 들어가는 위치가 다르게 보이고, [React.createElement API](https://ko.reactjs.org/docs/react-api.html#createelement)에서 세번째 인자를 children로 넣어주고 있어서 두번째 인자에 한번에 `...props`로 보내도 괜찮을까 싶었는데 실행시켜 보니까 잘되네요,,,,?,,,,,👉👈  

요것이 이해가 안되는 건 제가 JS문법 기초가 부족해서일까요,,,🥲  어떻게 이해하면 좋을지 조언해주시면 매우매우 감사하겠습니다,,, 🙏

Reply:
className에 아무것도 안넘긴다면 `undefined`: true가 되고 있었군요...! 수정해주신대로 하면 classnames가 알아서 처리해주니 문제 없겠네요! 감사합니다 😆

Reply:
제 로컬에 설치된 코드 스펠링 체크해주는 익스텐션 `cspell`에서 밑줄뜨면 별 생각없이 수정했는데, 필요할 경우 `user dictionary`에 추가하거나 `cspell.json`에서 설정을 작성자가 필요한대로 바꿀 수 있다는 사실을 알게되었습니다! ([참고문서](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)) `classnames`외에도 추가할 단어들이 있을 것 같고, 이 설정 파일도 push해놓는게 좋을 것 같아서 cspell.json에서 관리해보려고 합니당! 에디터에 `cspell`때문에 PROBLEM 몇십 개 줄줄이 뜨곤 했는데...(못본 척했지만...🥲) 이제  없앨 수 있게되었네요~~~😆